### DISCUSSION TITLE
React.lazy() doesn't work on static pages

### DISCUSSION THREAD

**Discussion by mimiqkz** [vercel/next.js#27344] [Help]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

14.17.3

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Azure

### Describe the Bug

Build error upon trying to use react.lazy()
According to [Nextjs](https://nextjs.org/docs/advanced-features/automatic-static-optimization), pages that doesn't have  getServerSideProps and getInitialProps in the page are static.
So I have a static page with `getStaticProps` and I try to use react.lazy() on it, then I get the error. Which doesn't make any sense.

### Expected Behavior

react.lazy() should be able to work in a static page

### To Reproduce

```
import {Suspense, lazy} from 'react'
import {GetStaticProps} from 'next'

const A = lazy(() => import('../../components/A'))
const B = lazy(() => import('../../components/B'))

export const getStaticProps: GetStaticProps = async () => {
  ...get some data here
  return {
    props: {
      data
    },
  }
}

interface Props {
  data: DataProps
}

export default function Index({
  data
}: Props) {

  return (
    <>
      <A data={data} />
      <B />
    </>
  )
}

```

**Comment by awareness481** [comment]
Static pages are still rendered on the server, so `React.lazy` won't work (because it is only supported on the client).

You can use `next/dynamic` instead https://nextjs.org/docs/advanced-features/dynamic-import

**Comment by mimiqkz** [comment]
@awareness481  I never thought static still rendered on the server so that's new for me. Thank you for the information :) Then in what cases are ok to use React.lazy?

**Reply by awareness481** [reply]
You can't use `React.lazy` in Next.js. The React team is planning to add support for `React.lazy` on the server at some point in React 18, so only then you could use `React.lazy` in Next.js :P

**Reply by bjarnivordur** [reply]
@awareness481 Can `React.lazy` be used in Next.js when the app is exported to static HTML with `next export`?
